### DISCUSSION TITLE
Rego 3.2 beta 3 & Ranger

### DISCUSSION THREAD

**Discussion by PatriceGendreau** [regolith-linux/regolith-desktop#1038] [General]
My favorite file manager "Ranger" cannot be launch with Ilia anymore... I can launch it only by terminal
It worked in 3.1
Thanks

**Comment by peterrus** [comment]
@PatriceGendreau Which version of Ubuntu are you running? (If Ubuntu)

**Reply by PatriceGendreau** [reply]
I use Ubuntu 24.04
The Ilia RANGER menu worked perfectly before Beta 3
Thanks

**Reply by PatriceGendreau** [reply]
PS I have ranger.desktop in /usr/share/applications/ranger.desktop
####
[Desktop Entry]
Type=Application
Name=ranger
Comment=Launches the ranger file manager
Icon=utilities-terminal
Terminal=true
Exec=ranger
Categories=ConsoleOnly;System;FileTools;FileManager
MimeType=inode/directory;
Keywords=File;Manager;Browser;Explorer;Launcher;Vi;Vim;Python

**Reply by PatriceGendreau** [reply]
Ranger IS in the Ilia menu
*ranger must be run from terminal*
It worked before the last update
Thanks

**Comment by peterrus** [comment]
If you manually run `ilia` from your terminal, do you see Ranger in the list then?

**Comment by kgilmer** [comment]
Running `ilia` from the command line and launching `ranger` from the apps dialog yields:

```shell
~$ ilia -p apps
Running as unit: run_ilia_ranger.desktop_43934d87.scope; invocation ID: 44fba3fd267d4fa5b82ca8b9a94ee65f
~$ Error: Must run ranger from terminal
```

I notice some fields in the Desktop file "Terminal=true" and "Categories=ConsoleOnly". I'm guessing that there is some specific handling `ranger` requires (specifically, to be launched from the context of an active terminal session) that `ilia` is not handling.  

I will look into handling these fields in a future update to `ilia`.  In the short term, a few options:

1. running `ranger` from the "commands" or "terminal" view of `ilia` works for me
2. create a custom desktop file in your user directory and have it exec the terminal instead of exec ranger directly (untested)
3. modify the `ranger` Desktop file to do the same as option 2

**Reply by kgilmer** [reply]
tracking issue: https://github.com/regolith-linux/ilia/issues/92

**Reply by PatriceGendreau** [reply]
Thanks Ken  now I use *shift-win-space* to run Ranger
I'll try your suggestions.....
What surprises me is that it worked perfectly before 3.2.....
Thanks again

**Reply by PatriceGendreau** [reply]
I have a SOLUTION !!!

in ranger.desktop
I change 
*exec=ranger*
with
*exec=gnome-terminal -- ranger*
and IT WORKS
Thanks
